chapters/arrays.tex

```diff
@@ -55,6 +55,7 @@ \section{Array Declarations}\label{array-declarations}
 \item
   Binding equations of the form \lstinline!x1 = x2! are allowed for arrays independent of whether the index types of dimensions are subtypes of \lstinline!Integer!, \lstinline!Boolean!, or enumeration types.
 \end{itemize}
+Operations constructing arrays (e.g., matrix multiplication, indexing, array constructors) only generate arrays index by \lstinline!Integer!.
```

Comment:
It seems very strange to me that matrix multiplication would turn enumeration-indexed dimensions into `Integer`-indexed:
```
type EA = enumeration(…);
type EB = enumeration(…);
type EC = enumeration(…);
Real[EA, EB] x;
Real[EB, EC] y;
EA a;
EB c;
Real z = (x * y)[a, c]; // Correct way to index product, right?!
```

Reply:
The enumeration-indexed dimensions are not well supported; so it is even unclear if `x*y` is legal etc. In general, I see some uses of enumeration-indexing - but currently it looks like something bolted-on, and I'm not if we have the resources to integrate it that fully. (BTW: `Real z=x[a,:]*y[:,c];` is, of course, an alternative for getting that element - which may be more efficient.)